tmc-plugin/src/fi/helsinki/cs/tmc/model/NbTmcSettings.java

```diff
@@ -153,9 +156,9 @@ public boolean isSavingPassword() {
 
     @Override
     public String getTmcMainDirectory() {
-        String path = settings.get(PREF_PROJECT_ROOT_DIR, null);
+        Path path = Paths.get(settings.get(PREF_PROJECT_ROOT_DIR, null));
         if (path != null) {
-            return path;
+            return path.toString();
```

Comment:
Not necessarily. I'd suspect we need to do something like
```Java
String target = settings.get(PREF_PROJECT_ROOT_DIR, null);
if (target != null) {
  return Paths.get(target);
} else {
  return ProjectMediator.getDefaultProjectRootDir();
}
```
to preserve the current functionality.